build_projects/dotnet-host-build/CompileTargets.cs

```diff
@@ -459,12 +459,14 @@ public static BuildTargetResult RestoreLockedCoreHost(BuildTargetContext c)
         {
             var hostVersion = c.BuildContext.Get<HostVersion>("HostVersion");
             var lockedHostFxrVersion = hostVersion.LockedHostFxrVersion.ToString();
+            var lockedHostVersion = hostVersion.LockedHostVersion.ToString();
             string currentRid = HostPackageSupportedRids[c.BuildContext.Get<string>("TargetRID")];
             string framework = c.BuildContext.Get<string>("TargetFramework");
 
             string projectJson = $@"{{
   ""dependencies"": {{
-      ""Microsoft.NETCore.DotNetHostResolver"" : ""{lockedHostFxrVersion}""
+      ""Microsoft.NETCore.DotNetHostResolver"" : ""{lockedHostFxrVersion}"",
+      ""Microsoft.NETCore.DotNetHost"" : ""{lockedHostVersion}"",
```

Comment:
I presume we are explicitly adding reference to Dotnethost here since it is no longer in the transitive closure of HostResolve, right? If so, it will be good to add a comment indicating the same.